### ISSUE TITLE
AssertionError when .sass-lint.yml or package.json missing

### ISSUE THREAD

**Issue by nene** [sasstools/sass-lint#563]
I would expect sass-lint to work by just running:

    $ sass-lint

Or at least it should provide a helpful error message describing which required options I'm missing. Instead it crashes:

```
assert.js:89
  throw new assert.AssertionError({
  ^
AssertionError: missing path
    at Module.require (module.js:364:3)
    at require (module.js:385:17)
    at module.exports (/usr/local/lib/node_modules/sass-lint/lib/config.js:73:12)
    at Function.sassLint.getConfig (/usr/local/lib/node_modules/sass-lint/index.js:25:10)
    at Function.sassLint.lintFiles (/usr/local/lib/node_modules/sass-lint/index.js:166:18)
    at detectPattern (/usr/local/lib/node_modules/sass-lint/bin/sass-lint.js:16:18)
    at Object.<anonymous> (/usr/local/lib/node_modules/sass-lint/bin/sass-lint.js:87:3)
    at Module._compile (module.js:435:26)
    at Object.Module._extensions..js (module.js:442:10)
    at Module.load (module.js:356:32)
```

So I follow the manual and create `.sass-lint.yml` and run it again:

    $ sass-lint

Again it crashes with the same stack trace as before.

So I go and look at this `config.js` file at line 73. Turns out it tries to read in `package.json`. Apparently that's a hidden feature implemented in #47, but documented nowhere.

So I create an empty `package.json` file consisting of just `{}` to please sass-lint.

Now finally I'm able to run it:

    $ sass-lint

But I see no errors, it appears to lint just fine, although I purposefully created a very buggy SCSS file. OK, I'll try to introduce a syntax error. I'll run the linter again, and it crashes:

```
/usr/local/lib/node_modules/sass-lint/index.js:250
    throw new Error(errorCount.count + ' errors were detected in \n- ' + errorCount.files.join('\n- '));
    ^

Error: 1 errors were detected in 
- test.scss
    at Function.sassLint.failOnError (/usr/local/lib/node_modules/sass-lint/index.js:250:11)
    at detectPattern (/usr/local/lib/node_modules/sass-lint/bin/sass-lint.js:27:10)
    at Object.<anonymous> (/usr/local/lib/node_modules/sass-lint/bin/sass-lint.js:87:3)
    at Module._compile (module.js:435:26)
    at Object.Module._extensions..js (module.js:442:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:311:12)
    at Function.Module.runMain (module.js:467:10)
    at startup (node.js:134:18)
    at node.js:961:3
```

That error doesn't really help me either. I search through the issues in this repository, and find #500, from where I finally learn that I need to run:

    $ sass-lint --no-exit --verbose

The [CLI docs](https://github.com/sasstools/sass-lint/tree/develop/docs/cli) don't explain any of this. There's no way I would have guessed that I need to use `--verbose` option. Issue #500 mentions that there are plans to improve this in 2.0, but I really suggest documenting the current behavior.

Using sass-lint 1.5.1

**Comment by DanPurdy** [comment]
Hi @nene I completely agree with you. The CLI and sass-lint itself had grown at a very quick pace and the amount of usage it was getting so early on kind of took us by surprise, in our efforts to get as much functionality and usability into the tool as soon as possible when we started this I think we kind of neglected some of the documentation and at the same time maybe took some decisions regarding the implementation which we now would change.

The CLI itself is due an overhaul for sass-lint v2 but in the meantime I've updated our README in PR #586  to provide a little more information straight off the bat about how to use the CLI.

I've also put PR #588 in to fix the assertion error you see if not specifying a config file. Both of these we've known about for a while but you know time is one of those things we struggle to find recently!

Anyway, hopefully the changes should see issues like yours become less frustrating from a user point of view and with the added test for this missing config error im sure we shouldn't see that raise its head again.

Thanks
